src/main/java/frc/robot/subsystems/DriveSubsystem.java

```diff
@@ -240,18 +243,30 @@ public void drive(double xSpeed, double ySpeed, double rot, boolean fieldRelativ
 
       xSpeedCommanded = m_currentTranslationMag * Math.cos(m_currentTranslationDir);
       ySpeedCommanded = m_currentTranslationMag * Math.sin(m_currentTranslationDir);
-      m_currentRotation = m_rotLimiter.calculate(rot);
+      m_rotationCommanded = m_rotLimiter.calculate(rot);
 
     } else {
       xSpeedCommanded = xSpeed;
       ySpeedCommanded = ySpeed;
-      m_currentRotation = rot;
+      m_rotationCommanded = rot;
     }
 
     // Convert the commanded speeds into the correct units for the drivetrain
     double xSpeedDelivered = xSpeedCommanded * DriveConstants.kMaxSpeedMetersPerSecond;
     double ySpeedDelivered = ySpeedCommanded * DriveConstants.kMaxSpeedMetersPerSecond;
-    double rotDelivered = m_currentRotation * DriveConstants.kMaxAngularSpeed;
+    double rotDelivered = 0.0;
+    if (DriverStation.isTeleopEnabled()) {
+      intendedRotation = intendedRotation
+          .rotateBy(new Rotation2d((-m_rotationCommanded) * ((timeOfLastLoop - System.currentTimeMillis()) / 1000)
```

Comment:
There are a couple of algebraic changes possible here which may make it easier to read by other coders.

The expression `timeOfLastLoop - System,currentTimeMillis()` will always be negative or zero.  That explains to me why there is a minus sign before `m_rotationCommanded`.  It would be slightly cleaner to take out the negatives.

Also, `System.currentTimeMillis()` always returns a long integer, so it's a little odd that the `timeOfLastLoop` variable is double floating point number.  You could change the variable to a `long` value as long as you changed the denominator to a floating point number (i.e `1000.0`).

Reply:
Fixed